test/unit/bigint-encoder_test.js

```diff
@@ -0,0 +1,115 @@
+import { encodeBigIntFromBits, formatIntName, sliceBigInt } from '../../src/bigint-encoder';
+
+describe('encodeBigIntWithPrecision', function () {
+  it(`encodes values correctly`, () => {
+    const testCases = [
+      // i64
+      [[0], 64, false, 0n],
+      [[-1], 64, false, -1n],
+      [['-15258'], 64, false, -15258n],
+      [[-0x8000000000000000n], 64, false, -0x8000000000000000n],
+      [[0x7FFFFFFFFFFFFFFFn], 64, false, 0x7FFFFFFFFFFFFFFFn],
+      [[1, -2147483648], 64, false, -0x7FFFFFFFFFFFFFFFn],
+      [[-1, -1], 64, false, -1n],
+      [[-2, 2147483647], 64, false, 9223372036854775806n],
+      [[345, -345], 64, false, -0x158fffffea7n],
+      // u64
+      [[0], 64, true, 0n],
+      [[1n], 64, true, 1n],
```

Comment:
Given the `parts.length === 0` is a special case that handles signed stuff differently, it would be nice to have a test case for that specifically:
```js
[[-1n], 64, true, /*??*/],
```

Reply:
I think this should be solved by the loop rewrite discussion above